bilge-impl/src/bitsize_internal.rs

```diff
@@ -237,3 +279,48 @@ fn generate_common(ir: ItemIr, arb_int: &TokenStream) -> TokenStream {
         }
     }
 }
+
+/// generate std::fmt impls - currently only std::fmt::Binary
+fn generate_enum_fmt_impls(variants: Iter<Variant>, enum_name: &Ident, arb_int: &TokenStream, bitsize: BitSize) -> TokenStream {
+    let to_int_match_arms = generate_to_int_match_arms(variants, enum_name, bitsize, arb_int);
+    
+    let body = if to_int_match_arms.is_empty() {
+        quote! { Ok(()) }
+    } else {
+        quote! {
+            let value = match &self {
+                #( #to_int_match_arms )*
+            };
+            write!(f, "{:0width$b}", value, width = <#enum_name as Bitsized>::BITS)
+        }
+    };
+
+    quote! {
+        impl core::fmt::Binary for #enum_name {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                #body
+            }
+        }
+    }
+}
+
+/// generates the arms for an (infallible) conversion from an enum to the enum's underlying arbitrary_int
+fn generate_to_int_match_arms(
```

Comment:
is the only difference that this doesn't do `from_int_match_arm`? Then I would rather merge it with the others, but oh right let's wait until after #44.